conans/client/command.py

```diff
@@ -748,11 +748,12 @@ def user(self, *parameters):
         parser.add_argument("name", nargs='?', default=None,
                             help='Username you want to use. '
                                  'If no name is provided it will show the current user.')
-        parser.add_argument("-p", "--password", help='User password. Use double quotes '
-                            'if password with spacing, and escape quotes if existing')
         parser.add_argument("--remote", "-r", help='look in the specified remote server')
         parser.add_argument('-c', '--clean', default=False,
                             action='store_true', help='Remove user and tokens for all remotes')
+        parser.add_argument("-p", "--password", nargs='?', const="", type=str,
+                            help='User password. Use double quotes if password with spacing, and escape quotes if '
+                                 'existing. Request for user password without expose it security data when empty')
```

Comment:
Maybe something like this is more clear for the help "With an empty `-p` value, it requests the password interactively (not exposed)"